[New New UI] Disable blurred bg behind image previews, hide post separators and bring back features from v2 UI

bmndc opened the pull request:
Follows up with #109. Please see each commit for changelog.

@joelacus I've backported Create Post Title and Body font size & font weight change feature to Old New UI, for those who landed on something like https://www.reddit.com/web/r/polls/submit (see 4896576). I think it'd be great that the very next 1.14.0 keep this and other Old New UI features, and then we make a new branch to preserve 1.14.0 in case https://github.com/dauphine-dev/returnNewReddit becomes a thing :)

joelacus commented:
This is quite a big PR, you've done a lot of work, thanks! I'll try to review and merge it by this weekend :)

I didn't realise any pages using the "Old New" version were left. I was planning to release RE version 2.0.0 after 1.14.0 and rip out all the code for "Old New UI" :/. I suppose I could keep some styling features compatible for the time being.

Even if this extension https://github.com/dauphine-dev/returnNewReddit becomes popular, it only seems to restyle the latest Reddit version, so the "Old New UI" features won't work with it. We could perhaps ensure it doesn't conflict with RE though if people want to use both extensions.

bmndc commented:
@joelacus you might really want to keep it around for a bit longer: https://www.reddit.com/r/ReturnNewReddit/comments/1hwkgz6/and_so_it_returns/

I'm refining some current features to work with the new ones; will take some notes so you can know what changed.

joelacus commented:
I had to push a bug fix. I resolved the conflicts though.
Reddit removed the "#main-content" id from \<main>.

bmndc commented:
@joelacus thanks—your commit scared me a bit, but it's all good haha

bmndc commented on src-webpack/src/common/content/popup_listener.js:
WebStorm keeps warning me of not using strict variable comparison (`===`), and throughout my testing changing settings seem to be still fine, so I keep them in.

bmndc commented on src-webpack/src/common/content/tweak_loader.js:
[New New UI] Wait until the subreddit page or the post page has loaded, then apply tweaks to remove paddings from the subreddit rule list, or apply custom border radius to the comment composer textarea.

I didn't use loadBorderRadiusAmount() here as it would result in a small delay & <style> tag being inserted repeatedly; although, in hindsight, I should have make the code in `addBorderRadiusToShadowRootElements()` condition-based... for now, it'll be invalid when `var(--re-theme-border-radius)` has not been set

bmndc commented on src-webpack/src/common/content/tweaks/background/custom_background.js:
![image](https://github.com/user-attachments/assets/4f15daa0-d202-4882-9b3f-20a5bb805291)

- Direct CSS padding and border radius directly onto `#main-content` rather than `shreddit-post` + comment composer + `shreddit-comment-tree`, which creates an unified and Old New UI-like post view.
- Do not apply custom background to post pane in mod queue, which would look really weird and is locked behind layers of shadow DOMs.

bmndc commented on src-webpack/src/common/content/tweaks/font/resize_font.js:
![](https://github.com/user-attachments/assets/9ac62d14-50d6-47e9-8b86-9f4b3ac656dd)

Resizing text works with h1 elements, comment composer, x-posted posts, posts in mod queue and Create Posts view in Old New UI. Small changes to line-height: 1.4 for titles, 1.5 for contents, 1.6 for comments

bmndc commented on src-webpack/src/common/content/tweaks/hide_elements/hide_blurred_media_background.js:
![](https://github.com/user-attachments/assets/decec9f3-4842-41cd-ae2c-a4ed8df9f88c)

bmndc commented on src-webpack/src/common/content/tweaks/hide_elements/hide_header_bar.js:
Use Reddit's custom CSS variables to remove the top padding reserved for the header bar.

bmndc commented on src-webpack/src/common/content/tweaks/hide_elements/hide_post_divider.js:
![](https://github.com/user-attachments/assets/a2104665-1748-4396-84c3-e5ddd7118826)

Without the post seperators, I had to add some margins and paddings to each shreddit-post to make sure they're spaced. Post Gap is customisable and the slider is under the Hide Post Divider toggle.

bmndc commented on src-webpack/src/common/content/tweaks/hide_elements/hide_side_menu.js:
* Move the back button inside the main-content container. For certain zoom levels, it might overflow to the left side of the screen.
* Fix an issue in which grid container doesn't restore itself when Reddit switches to mobile view. If Hide the Side Menu/Side Menu Toggle Button is on and the viewport is less than 1200px, Reddit will attempt to "hide" the side menu itself, which moves #main-content to first grid column and causes grid-template-column to break.

bmndc commented on src-webpack/src/common/content/tweaks/hide_elements/hide_sidebar.js:
Add function to hide the information sidebar in Custom Feed view.

bmndc commented on src-webpack/src/common/content/tweaks/hide_elements/side_menu_toggle_button.js:
Same as hide_side_menu.js. Do not permanently set --re-side-menu-width to 0.

bmndc commented on src-webpack/src/common/content/tweaks/productivity/show_post_author.js:
Move the post author between the subreddit title and the time to replicate the Old New UI style. Display on custom feeds and custom feed search results.

![](https://github.com/user-attachments/assets/62534fd3-97c0-4484-b0e5-e01e37ddc6dc)

bmndc commented on src-webpack/src/common/content/tweaks/productivity/show_post_flair.js:
Add flairs to posts in Custom Feed view. Fix an issue where flair containers are shown on posts with no flairs

bmndc commented on src-webpack/src/common/content/tweaks/resize_elements/expand_content.js:
Minor fixes to accommodate the Resize Main Container feature

bmndc commented on src-webpack/src/common/content/tweaks/style/border_radius.js:
Only change the corner radius on known elements which are outside shadow DOMs. Attempt to apply corner radius to elements inside shadow DOMs, such as the post comment composer

bmndc commented on src-webpack/src/common/content/tweaks/style/old_new_ui.js:
```diff
@@ -0,0 +1,205 @@
+/**
```
![](https://github.com/user-attachments/assets/77c65b45-0fd6-48df-bb50-b28848330fdd)

bmndc commented on src-webpack/src/common/content/tweaks/style/old_new_ui.js:
```diff
@@ -0,0 +1,205 @@
+    });
+}
+
+/**
```
![](https://github.com/user-attachments/assets/67f56217-6c40-4712-a50b-b4d583231508)

bmndc commented on src-webpack/src/common/content/tweaks/style/old_new_ui.js:
```diff
@@ -0,0 +1,205 @@
+    });
+}
+
+/**
```
Currently this only sets the header bar height to 48px and removes padding from the side menu. Items in the side menu are dynamically rendered under layers of shadow DOMs, which makes styling very difficult.

TODO: this menu?

![](https://github.com/user-attachments/assets/ecd34624-7228-49c5-b64c-d1b8170ed0e4)

bmndc commented on src-webpack/src/common/content_first/functions/load_styles.js:
Minor styling tweaks as we change how Show Post Author is shown

bmndc commented on src-webpack/src/common/content/tweaks/style/override_theme_colours.js:
In New New UI, Reddit intends the post in post view to cover the browser view, with `--color-neutral-background` set as background color. As such, do not change the colour of `shreddit-post` + `shreddit-comment-tree` separately, but use custom CSS variables Reddit provided to set the colour of the entire post view. One can use Custom Background to replicate the older behaviour of this setting.

Only add post borders to posts in feeds and not in post view, and use Reddit's custom CSS variables to set the colour of links.

BUG: Setting the alpha level on Post Background may cause the comment nest lines (.threadline) to act weird.

![](https://github.com/user-attachments/assets/503ab9bc-07aa-4d7a-b5e1-16f15fdb7660)

bmndc replied:
@joelacus that's not the issue. I mean, this is what Reddit intends `.threadline` in New New UI to behave:

![](https://github.com/user-attachments/assets/a6a5b1f1-0614-440f-b011-f777129f1d32)

Setting `--color-neutral-background: transparent` on `shreddit-comment` as you discussed would expose some new lines leading to nowhere:

![](https://github.com/user-attachments/assets/d5700022-f549-4d6e-ba2a-fbceb469f138)

Some may like it, but I'm not "bringing back" this Old New UI style of comment nest lines at the moment.

joelacus replied:
![image](https://github.com/user-attachments/assets/e9080881-b160-4892-9172-a0b3ceace909)

Setting this fixes the overlapping colours, but the lines going nowhere show regardless for me :s
```
shreddit-comment {
    --color-neutral-background: transparent !important;
}
```

bmndc replied:
I think it might be okay to keep this behaviour until I can figure out how to do this in the Old New UI manner...

joelacus replied:
@bmndc I fixed this issue in ce5c3d1. The issue is that the line is always there, and is just covered by a block the same colour as the post background, but with translucent colours, this doesn't really work. I've made it so that If it detects an rgba colour for the post background, it will get the height of that block, and shorten the thread line, for all comments.

bmndc replied:
I was considering getting rid of the rounded corners, but that also works too. Seems to function fine for me so far.

joelacus commented:
Thanks for commenting on the changes, you can resolve the conversations now if there's nothing more to add.

Also, could you give me an updated changelog please so I can copy it into the extension.

Thanks for all your work on this, I really appreciate it! :) I just have a few more files to review and then I'll merge.

bmndc commented:
@joelacus 

```
* Added options to hide post separators in feed view and adjust gaps between posts
* Added an option to hide the background blur behind image previews of posts in feed view
* Added an option to hide the custom feed sidebar
* Show Post Flairs on the Home Feed now also shows post flairs on custom feeds
* Show Post Author now also shows authors on custom feeds and custom feed search results
* Added an option to stretch the subreddit banners to take the entire width of browser window
* Added an option to tweak the subreddit rule list to be more compact and resemble Old New UI
* (experimental) Added an option to tweak the header and side menu to be more compact
* (experimental) Added an option to resize the entire content container instead of just the feeds
* Fixed an issue where post flair container may show on posts with no flairs when Show Post Flairs on the Home Feed is on
* Fixed an issue where the grid container may not restore itself when Reddit switches to mobile view (viewport < 1200px)
* Back button now remains inside the post container as it might be obstructed at certain zoom levels
* Show Post Author now resembles the Old New UI post author style
* Custom Background no longer applies to post pane in moderation queue :(
* Hide the Header Bar now edits Reddit's CSS variables to remove the top padding reserved for the header bar
* Resizing fonts now works with h1 elements, comment composer, crossposted posts, posts in moderation queue and Create Posts view in Old New UI. Small changes to line heights
* Use Reddit's CSS variables to set post and comment link colour
* Post borders now only show in feed view and no longer appear in post view
* Setting the background colour of posts now uses Reddit's native CSS variables and apply to the entire post view container. It is possible to replicate the older behaviour by setting a custom theme background.
* Small style tweaks to Theme Corner Radius, hover cards and other elements throughout
```

joelacus commented:
> Small favour, could you help with the implementation of Hide the Custom Feed Sidebar on Old New UI? It's supposed to only target /user/*/m/ URLs, but I'm unable to get it working without hiding the sidebar of other pages.

I can't access custom feeds on the old new UI, so I don't know if there's any element you can use to identify the page, but you could run something like this so it only runs the function on a page if the URL contains `/user/*/m/` :

```
const link = window.location.href;
if (/\/user\/[^\/]+\/m\//.test(link)) {
    // add stylesheet
} else{
    // remove stylesheet
}
```

bmndc commented:
Looking into it really quick before you merge the PR ;-)

joelacus commented:
I noticed you also optimised some stuff like the event listeners in `inputs_hide_elements.js`, thanks :)

bmndc commented:
I was considering getting rid of the rounded corners, but that also works too. Seems to function fine for me so far.
On Feb 13, 2025, 1:51 PM +0700, Joelacus ***@***.***>, wrote:

>
> @joelacus commented on this pull request.
> On src-webpack/src/common/content/tweaks/style/override_theme_colours.js:
> @bmndc I fixed this issue in ce5c3d1. The issue is that the line is always there, and is just covered by a block the same colour as the post background, but with translucent colours, this doesn't really work. I've made it so that If it detects an rgba colour for the post background, it will get the height of that block, and shorten the thread line, for all comments.
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID: ***@***.***>
